packages/react-native-bpk-component-calendar/src/ios/CalendarBridge/RCTBPKCalendarManager.m

```diff
@@ -107,4 +111,83 @@ - (void)calendar:(RCTBPKCalendar *)calendar didChangeDateSelection:(NSArray<BPKS
     calendar.onDateSelection(@{@"selectedDates": dateArray});
 }
 
+- (RCTBPKColorBucket *_Nullable)colorBuckedForDate:(NSDate *)date
+                                      colorBuckets:(NSArray<RCTBPKColorBucket *> *)colorBuckets {
+    for (RCTBPKColorBucket *bucket in colorBuckets) {
+        switch (bucket.days.matcherType) {
+        case RCTBPKDateMatcherTypeRange:
+            if ([RCTBPKCalendarDateUtils date:date isAfterDate:bucket.days.dates[0]] &&
+                [RCTBPKCalendarDateUtils date:date isBeforeDate:bucket.days.dates[1]]) {
+                return bucket;
+            }
+            break;
+        case RCTBPKDateMatcherTypeBefore:
+            if ([RCTBPKCalendarDateUtils date:date isBeforeDate:bucket.days.dates[0]]) {
+                return bucket;
+            }
+            break;
+        case RCTBPKDateMatcherTypeAfter:
+            if ([RCTBPKCalendarDateUtils date:date isAfterDate:bucket.days.dates[0]]) {
+                return bucket;
+            }
+            return nil;
+        case RCTBPKDateMatcherTypeAny:
+            for (NSDate *bucketDate in bucket.days.dates) {
+                if ([bucketDate isEqualToDate:date]) {
+                    return bucket;
+                }
+            }
+            break;
+        }
+    }
+    return nil;
+}
+
+- (UIColor *)calendar:(BPKCalendar *)calendar fillColorForDate:(NSDate *)date {
+    RCTBPKCalendar *rctCalendar = (RCTBPKCalendar *)calendar;
+    if (rctCalendar != nil) {
+        RCTBPKColorBucket *colorBucket = [self colorBuckedForDate:date colorBuckets:rctCalendar.rct_colorBuckets];
+
+        if (colorBucket) {
+            return colorBucket.color;
+        }
+    }
+    return BPKColor.clear;
+}
+
+- (UIColor *)calendar:(BPKCalendar *)calendar titleColorForDate:(NSDate *)date {
+    NSAssert([calendar isKindOfClass:RCTBPKCalendar.class],
```

Comment:
All asserts need to be coupled with an if using the reversed condition for production since asserts do nothing in production builds. Without it this code would crash at runtime in production builds due to `colorBuckedForDate:colorBuckets:` being unrecgonized

Reply:
Done